WordPress/gutenberg
# Use page list instead of placeholder as fallback

Closes #42563

- [x] Default to a page list block when the navigation block is empty and no menus exist
  - [x] Add test for this
  - [x] Update tests that assume empty nav defaults to placeholder
    - [x] Update `selectClassicMenu` to pull the block toolbar instead of placeholder
  - [x] Update the front end to default to a page list block when the navigation block is empty and no menus exist
- [x] Show "empty placeholder" when deleting the selected menu from the block
  - [ ] Add test for this
- [x] Default to the most recently created menu if multiple menus exist
  - [x] Add test for this
- [x] Default to the only menu if only one menu exists
  - [x] Add test for this
- [x] 🐛 deleting the fallback page list block results in a menu being created, i think this is correct **for now**
- [x] 🐛 inserting two blocks when there are no menus (fallback on page list) automatically creates a new menu
- [ ] 🐛 responsive navigation blocks don't allow the fallback page list to be selected via mouse click, , opened #43227



## What?


This makes the navigation block default to a page list in the editor when loading an empty navigation block (one that has no uncontrolled blocks nor any ref).

## Why?


We do this so that the editor respects the fallback of the front end, which is a page list.

## How?


We use a similar technique to the one used in the pattern block which does not add the pattern before it is edited.

## Testing Instructions

1. Use a theme that has a template part with an empty navigation block (e.g. Vivre)
2. Load the template part in the site editor
3. Make sure you delete ALL your navigation menus
4. You should see a page list block for navigation

## Screenshots or screencast 

N/A
